gemrb/GUIScripts/pst/GUIREC.py

```diff
@@ -818,7 +818,7 @@ def AcceptLevelUp():
 	GemRB.SetPlayerStat (pc, IE_HITPOINTS, HPGained+oldhp)
 	#increase weapon proficiency if needed
 	if WeapProfType!=-1:
-		GemRB.SetPlayerStat (pc, WeapProfType, CurrWeapProf + WeapProfGained )
```

Comment:
If you don't like the addition, just set up WeapProfType properly. After all it's read from a table, so it can be fixed in the data.

Reply:
I think it just reads weird when you look at it because of the name ```IE_PROFICIENCYBASTARDSWORD``` which is kind of just an alias for eg "IE_PROFICIENCYFISTS" if the definitions had PST based names. The result is exactly the same though, and this method is used various other places.